crates/ibc/src/core/ics03_connection/connection.rs

```diff
@@ -113,6 +112,17 @@ pub struct ConnectionEnd {
     delay_period: Duration,
 }
 
+impl TryFrom<ConnectionEnd> for Vec<u8> {
+    type Error = ConnectionError;
+
+    fn try_from(value: ConnectionEnd) -> Result<Self, Self::Error> {
+        let value = value
+            .encode_vec()
+            .map_err(ConnectionError::InvalidConnectionEnd)?;
+        Ok(value)
+    }
+}
+
```

Comment:
Also, we can keep them inside a private module, so use the benefit of this conversion internally. WDYT?
```rust
mod private {
use super::*;
-----
}

Reply:
I can see the value of having these conversions internally yep. But I'd do helper functions with different names; e.g. `ConnectionEnd::proto_encode_vec()` or something like that. Again, because I'd reserve the use of `TryFrom` only for when there's one way to do the conversion

Reply:
[bfe0004](https://github.com/cosmos/ibc-rs/pull/531/commits/bfe0004cadf558ff5c0fad84f6fdd75e7a0489f5)